sensor_control/battery_monitor/nodes/battery_monitor.py

```diff
@@ -0,0 +1,162 @@
+
+
+# Voltage threshold for the initial low voltage alarm
+# 22.1V (approximately 30%) is the recommended value
+battery_low = 22.1
+
+# Voltage threshold for the critically low voltage alarm
+# 20.6V (approximately 15%) is the recommended value
+battery_critical = 20.6
+
+# Voltage threshold for the total system kill alarm
+# 19.5V (approximately 5%) is the recommended value
+battery_kill = 19.5
+
+rospy.init_node('battery_monitor')
+
+
+class BatteryMonitor():
+
+    def __init__(self):
+
+        # Initialize the four receiving variables to max voltage
+        self.supply_voltage = [29.4, 29.4, 29.4, 29.4]
+
+        # Initialize the average voltage to max
+        self.voltage = 29.4
+
+        # Initialize the variables for handling averaging (in an attempt to eliminate outliers)
+        self.avg_count = [0, 0, 0, 0]
+        self.avg_value = [0, 0, 0, 0]
+
+        # The publisher for the averaged voltage
+        self.pub_voltage = rospy.Publisher("/battery_monitor", Float32, queue_size=1)
+
+        # Subscrives to the feedback from each of the four thrusters
+        rospy.Subscriber("/FL_motor/feedback", Feedback, self.add_to_fl)
+        rospy.Subscriber("/FR_motor/feedback", Feedback, self.add_to_fr)
+        rospy.Subscriber("/BL_motor/feedback", Feedback, self.add_to_bl)
+        rospy.Subscriber("/BR_motor/feedback", Feedback, self.add_to_br)
+
+        # Sets up the battery voltage alarms
+        alarm_broadcaster = AlarmBroadcaster()
+        self.battery_low_alarm = alarm_broadcaster.add_alarm(
+            name='battery_low',
+            action_required=False,
+            severity=2
+        )
+        self.battery_critical_alarm = alarm_broadcaster.add_alarm(
+            name='battery_critical',
+            action_required=True,
+            severity=1
+        )
+        self.battery_kill_alarm = alarm_broadcaster.add_alarm(
+            name='kill',
+            action_required=True,
+            severity=0
+        )
+
+    def add_to_fl(self, msg):
+        '''
+        Simple method to connect the FL thruster feedback subscriber to the add_voltage method
+        '''
+        self.add_voltage(msg, 0)
+
+    def add_to_fr(self, msg):
+        '''
+        Simple method to connect the FR thruster feedback subscriber to the add_voltage method
+        '''
+        self.add_voltage(msg, 1)
+
+    def add_to_bl(self, msg):
+        '''
+        Simple method to connect the BL thruster feedback subscriber to the add_voltage method
+        '''
+        self.add_voltage(msg, 2)
+
+    def add_to_br(self, msg):
+        '''
+        Simple method to connect the BR thruster feedback subscriber to the add_voltage method
+        '''
+        self.add_voltage(msg, 3)
+
+    def add_voltage(self, msg, thruster_id):
+        '''
+        Averages 42 values from each thruster's feedback (42 is a calculated answer to life, the universe, and everything)
+        '''
+        if (self.avg_count[thruster_id] != 42):
```

Comment:
At the refresh rate of the message, the rolling average will be cleared by the time the message refreshes; however, it will probably be a lot cleaner. I will make a combined rolling average in my next commit.